core/tests/puppeteer-acceptance-tests/utility-functions/blogPostAdminUtils.js

```diff
@@ -0,0 +1,132 @@
+
+
+module.exports = class e2eBlogPostAdmin {
+  page;
+  browserInstance;
+
+  async openBrowser() {
+    this.browserInstance = await new browser();
+    this.page = await (this.browserInstance).initialize();
+  }
+
+  async signInWithEmail(email) {
+    await (this.browserInstance).signInWithEmail(email);
+  }
+
+  async waitForPageToLoad(selector) {
+    await (this.page).waitForSelector(selector);
+  }
+
+  async goto(destination) {
+    await (this.browserInstance).goto(destination);
+  }
+
+  async createDraftBlogPostByTitle(draftBlogPostTitle) {
+    try{
+      await (this.browserInstance).clickOn("span", "NEW POST");
+    } catch {
+      await (this.browserInstance).clickOn("span", " CREATE NEW BLOG POST ");
+    }
+    await (this.browserInstance).type(blogTitleInput, draftBlogPostTitle);
+    await (this.browserInstance).type(blogBodyInput, "blog post test body content");
+    await (this.browserInstance).clickOn("span", " DONE ");
+    await (this.browserInstance).clickOn("span", "SAVE AS DRAFT", 500);
+    
+    await console.log("Successfully created a draft blog post!");
+    await this.browserInstance.goto(blogDashboardUrl);
+
+    // this.page = await this.browserInstance.browser.newPage();
+  }
+
+  async deleteDraftBlogPost() {
+    await (this.browserInstance).clickOn("span", "Delete", 100);
+    await (this.browserInstance).clickOn("button", " Confirm ");
+
+    console.log("draft blog post with given title deleted successfully!");
+  }
+
+  async deleteDraftBlogPostByTitle(draftBlogPostTitle) {
+    let draftBlogPostInfo = {
+      "draftBlogPostTitle": draftBlogPostTitle,
+      "deleteDraftBlogPost": this.deleteDraftBlogPost(),
+    }
+    await (this.page).evaluate(async(draftBlogPostInfo) => {
+      const allDraftBlogPosts = document.getElementsByClassName('blog-dashboard-tile-content');
+      for(let i = 0; i < allDraftBlogPosts.length; i++) {
+        let draftBlogPostTitle = allDraftBlogPosts[i].getElementsByClassName('e2e-test-blog-post-title')[0].innerText;
+        if(draftBlogPostTitle === draftBlogPostInfo.draftBlogPostTitle) {
+          allDraftBlogPosts[i].getElementsByClassName('e2e-test-blog-post-edit-box')[0].click();
+          await draftBlogPostInfo.deleteDraftBlogPost;
+          return;
+        }
+      }
+    }, draftBlogPostInfo);
+  }
+
+  async createNewBlogPostByTitle(newBlogPostTitle) {
+    try{
+      await (this.browserInstance).clickOn("span", "NEW POST");
+    } catch {
+      await (this.browserInstance).clickOn("span", " CREATE NEW BLOG POST ");
+    }
+    await (this.browserInstance).type(blogTitleInput, newBlogPostTitle);
+    await (this.browserInstance).type(blogBodyInput, "blog post test body content");
+
+    await (this.browserInstance).clickOn("div", thumbnailPhotoBox);
+    await (this.browserInstance).uploadFile('collection.svg');
+    await (this.browserInstance).clickOn("button", " Add Thumbnail Image ");
+    await (this.page).waitForTimeout(500);
```

Comment:
How is the value of 500 determined, and can we standardize it using something like "wait for event X to happen" instead?

Reply:
we need to wait for some time as we publish and then expect that if it exists.. the time required is for backend request to collect the data (newly created). Also, we can wait for the static components using the css/X-path selectors, so we need to wait particularly for some time (here I waited for 500ms) but this time depends on the internet speed, throttled network needs to wait for more time. we need to discuss more about this!

Reply:
Right, but in webdriverio there is this construct along the lines of "wait for XYZ to appear" before doing the next action. Can't we use a similar construct here rather than a numerical timeout? That way, even if the backend is slower than 500ms, this would still work.

See e.g. https://stackoverflow.com/questions/46135853/puppeteer-how-to-wait-until-an-element-is-visible

Reply:
Yes, we are using the waitForSelector() functions at every step in our tests, but suppose there are 'x' published blogs, we can search them using there classNames which is not unique.. so we cannot uniquely identify a single blogpost uniquely expect its Title, I'll be trying implementing a method inside the utilities that will be fulfilling this purpose! thanks!

Reply:
This is still not addressed? The "wait" should be until a particular condition is satisfied, not based on a hardcoded timeout.